extensions/amp-viqeo-player/0.1/amp-viqeo-player.js

```diff
@@ -0,0 +1,347 @@
+/* eslint-disable no-unused-vars */
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Deferred} from '../../../src/utils/promise';
+import {Layout, isLayoutSizeDefined} from '../../../src/layout';
+import {Services} from '../../../src/services';
+import {VideoEvents} from '../../../src/video-interface';
+
+import {assertAbsoluteHttpOrHttpsUrl, tryDecodeUriComponent} from '../../../src/url';
+import {dev, user} from '../../../src/log';
+import {fullscreenEnter, fullscreenExit, isFullscreenElement, removeElement} from '../../../src/dom';
+import {getData, listen} from '../../../src/event-helper';
+import {getIframe} from '../../../src/3p-frame';
+import {installVideoManagerForDoc} from '../../../src/service/video-manager-impl';
+
+/**
+ * @implements {../../../src/video-interface.VideoInterface}
+ */
+class AmpViqeoPlayer extends AMP.BaseElement {
```

Comment:
based on my testing, your video player `autoplay` regardless of `autoplay` attribute and does not adhere to AMP's autoplay UX.

Please make sure your place does not `autoplay` from your side. AMP managed `autoplay` automatically by calling `muted` and `play` actions and enfoces uniform UX (e.g. autoplay only when 50% and pause when 50% invisible).

Reply:
I've checked it